src/main/java/com/google/sps/data/friend_map/UserNode.java

```diff
@@ -0,0 +1,66 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.data.friend_map;
+
+import java.util.Collection;
+import com.google.common.collect.ImmutableSet;
+
+/**
+* Class to represent a User in the map representing all user friendships
+*/
+public class UserNode {
+
+  private final String userID;
+  private ImmutableSet<String> currentUserFriendIDs;
+
+  /**
+  * Constructs an instance of a UserNode
+  *
+  * @param userID The unique id number of each user (obtained from Facebook API)
+  * @param currentUserFriendIDs Immutable set of all of the IDs of a user's friends
+  */
+  public UserNode(String userID, Collection<String> currentUserFriendIDs) {
+    this.userID = userID;
+    this.currentUserFriendIDs = ImmutableSet.copyOf(currentUserFriendIDs);
+  }
+
+  public String getUserID() {
+    return this.userID;
+  }
+
+  public ImmutableSet<String> getCurrentUserFriendIDs() {
+    return this.currentUserFriendIDs;
+  }
+
+  @Override
+  public boolean equals(Object obj) {
+    if (!(obj instanceof UserNode)) {
+      return false;
+    }
+    UserNode otherUser = (UserNode) obj;
+    return this.userID.equals(otherUser.userID) &&
+      this.currentUserFriendIDs.equals(otherUser.currentUserFriendIDs);
```

Comment:
To check: Do you know what the `equals` in the `ImmutableSet` class does?

Reply:
I checked it out and it looks like ImmutableSet equals method would end up calling the Sets.equalsImp method (because the isHashCodeFast() method looks like it will go ahead ans return false?), and that will check if the sizes are the two sets are equal and if they contain all the same elements.
Should I still be using the ImmutableSets equals method here or is there a more efficient way to check these two sets?

Reply:
The "two sets are equal if they contain all the same elements" is where my concern lies. How does it determine that? I assume it's probably only doing memory address comparisons, which means each object will only be considered equal to itself, but not equal to a different object that contains the exact same data.

Reply:
It's using a containsAll method, and from my understanding, contains and containsAll methods use the equals method for the objects they're comparing. So I'd assume based on this that the String class's equals method would be used to compare the elements within these sets.